core/src/main/java/org/infinispan/interceptors/impl/TxInterceptor.java

```diff
@@ -341,10 +406,18 @@ public int size() {
                   }
                   return (int) size;
                }
+
+               private Map.Entry<K, V> toEntry(Object obj) {
+                  if (obj instanceof Map.Entry) {
+                     return (Map.Entry) obj;
+                  } else {
+                     return null;
```

Comment:
Again, do we expect non-null non-entry objects here?

Reply:
👍

Reply:
See above comment, but it has to do this :(